lib/system/excpt.nim

```diff
@@ -474,12 +474,34 @@ when defined(endb):
 when defined(cpp) and appType != "lib" and
     not defined(js) and not defined(nimscript) and
     hostOS != "standalone" and not defined(noCppExceptions):
+  proc verboseTerminate() {.importc: "__gnu_cxx::__verbose_terminate_handler", header: "<exception>".}
+  proc getTerminate: proc() {.noconv.}
+    {.importc: "std::get_terminate", header: "<exception>".} # C++11
+  var oldTerminate = getTerminate()
   proc setTerminate(handler: proc() {.noconv.})
     {.importc: "std::set_terminate", header: "<exception>".}
   setTerminate proc() {.noconv.} =
     # Remove ourself as a handler, reinstalling the default handler.
     setTerminate(nil)
 
+    if nil == currException:
+      when true:
+        writeToStdErr "egotCurrentException() -> nil\n"
+        {.emit: """
```

Comment:
This hack at least (!) needs a very good comment.

Reply:
There is no need for emit here, here is example how to do it: https://github.com/nim-lang/Nim/blob/devel/tests/exception/tcpp_imported_exc.nim

```nim
type 
  std_exception* {.importcpp: "std::exception", header: "<exception>".} = object

proc what(ex: std_runtime_error): cstring {.importcpp: "((char *)#.what())".}

try:
  myexception()
except std_exception as ex:
  echo $ex.what()
```

Reply:
Wouldn't we still need to "emit" a re-throw for `myexception()`?